Unity-Technologies/Graphics
# [SG2] First pass of numeric node UI controls

### Purpose of this PR

Split from https://github.com/Unity-Technologies/Graphics/pull/7191.

This PR adds working numeric node UI controls for the types:

- Float
- Int
- 3x3 Matrix
- 4x4 Matrix

These controls will automatically be added to node ports with `Usage.Static`.

![image](https://user-images.githubusercontent.com/10332426/155244534-eaf8dcdf-3ce8-4036-a73b-b16536fef6b4.png)

---
### Testing status

4 test nodes, pictured above, are included in this PR (eventually to be put in a category and excluded from production).

Note that typing into the numerical inputs can be difficult because the shader is currently recompiled with every change - go slow for best results.

- Test UI Int - returns a float4 with all components equal to the given input
  - Set the field to 0 and see that the result is dark.
  - Set the field to 1 and see that the result is bright.
  - Try entering a decimal value like 0.5 and see that it is not accepted.
  - Enter an expression and see that the field and preview update with the truncated result.

- Test UI Float - returns a float4 with all components equal to the given input
  - Set the field to 0 and see that the result is dark.
  - Set the field to 1 and see that the result is bright.
  - Set the field to 0.5 and see that the result is a mid gray.
  - Enter an expression and see that the field and preview update with the result.

- Test UI Mat 3 and 4 - returns a float4 using values from the diagonal of the matrix
  - Enter values into the diagonal (see non-zero fields in picture above) and see that, from left to right, they are used as red, green, and blue values in the preview below (bottom right of mat 4 is used as alpha but changes aren't visible).

---
### Comments to reviewers
